internal/cloudapi/v2/openapi.v2.yml

```diff
@@ -483,22 +483,30 @@ components:
         - architecture
         - image_type
         - repositories
-        - upload_request
+        - upload_options
       properties:
         architecture:
           type: string
           example: 'x86_64'
         image_type:
-          type: string
-          example: 'ami'
+          $ref: '#/components/schemas/ImageTypes'
         repositories:
           type: array
           items:
             $ref: '#/components/schemas/Repository'
         ostree:
           $ref: '#/components/schemas/OSTree'
-        upload_request:
-          $ref: '#/components/schemas/UploadRequest'
+        upload_options:
+          $ref: '#/components/schemas/UploadOptions'
+    ImageTypes:
+      type: string
+      enum:
+        - aws.ec2
+        - aws.s3
```

Comment:
I don't think we need to support S3 here, we should only ever produce AMIs in EC2.